pkg/metrics/exporter.go

```diff
@@ -49,19 +54,48 @@ func RegisterPrometheusExporter(registry *prom.Registry) (http.Handler, error) {
 // RegisterStackdriverExporter register a Stackdriver exporter to OpenCensus.
 // It will add Agones metrics into Stackdriver on Google Cloud.
 func RegisterStackdriverExporter(projectID string) (sd *stackdriver.Exporter, err error) {
-	// Default project will be used
+	instanceID, err := metadata.InstanceID()
+	if err != nil {
+		logger.WithError(err).Warn("error getting instance ID")
+		instanceID = unknown
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		logger.WithError(err).Warn("error getting zone")
+		zone = unknown
+	}
+	clusterName, err := metadata.InstanceAttributeValue("cluster-name")
+	if err != nil {
+		logger.WithError(err).Warn("error getting cluster-name")
+		clusterName = unknown
+	}
+
 	sd, err = stackdriver.NewExporter(stackdriver.Options{
 		ProjectID: projectID,
 		// MetricPrefix helps uniquely identify your metrics.
 		MetricPrefix: "agones",
+		Resource: &monitoredres.MonitoredResource{
+			Type: "container",
+			Labels: map[string]string{
+				"project_id":   projectID,
+				"instance_id":  instanceID,
+				"zone":         zone,
+				"cluster_name": clusterName,
+
+				// See: https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/
+				"namespace_id":   os.Getenv("POD_NAMESPACE"),
+				"pod_id":         os.Getenv("POD_NAME"),
+				"container_name": os.Getenv("CONTAINER_NAME"),
+			},
+		},
 	})
 	if err != nil {
 		return
 	}
 
 	// Register it as a metrics exporter
 	view.RegisterExporter(sd)
-	return
+	return sd, nil
```

Comment:
It's confusing to have some return paths use the blank return and have others pass the return values. I think the function is long enough that you should always return the explicit values and remove the named return values from the function declaration but that's just my personal preference.

Reply:
yes I agree, I wasn't also a fan of that naked return and hesitant to change it, now I'm convinced.